### ISSUE TITLE
[PATCH] Unhandled exception for incomplete XMLRPC

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10568]
|<img alt="Shawn's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Shawn reported|
|-|-|
|Trac ID|trac#872|
|Type|defect|
|Created|2005-01-25 21:45:07Z|

Attachments:

* [twisted.xmlrpclib.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/389/389b4f6ee5bd60bebd9d0708da23ba8b4134620b/bc46f4aa904b1e072d7decc9a63bc8a524ccf265.patch) (446 bytes) - added by Shawn on 2005-01-25 21:44:59Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__872 872
type__defect defect
reporter__Shawn Shawn
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__ 
time__1106689507000000 1106689507000000
changetime__1143334845000000 1143334845000000
version__ 
owner__ 
cc__exarkun cc__itamarst cc__Shawn
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="Shawn's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Shawn commented|
|-|-|

```
#!html
<pre>
Here is a small patch to handle xmlrpclib.ResponseError exceptions when
processing incomplete xmlrpc messages.  Here are the last few frames from the
traceback:

  File "/usr/lib/python2.3/site-packages/twisted/web/xmlrpc.py", line 271, in
handleResponse
    self.factory.parseResponse(contents)
  File "/usr/lib/python2.3/site-packages/twisted/web/xmlrpc.py", line 296, in
parseResponse
    response = xmlrpclib.loads(contents)
  File "/usr/lib/python2.3/xmlrpclib.py", line 1014, in loads
    return u.close(), u.getmethodname()
  File "/usr/lib/python2.3/xmlrpclib.py", line 740, in close
    raise ResponseError()
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

```
#!html
<pre>
Maintainer says you!  Should we apply this patch?
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `closed`|
|-|-|

(In [16453]) Fix t.w.xmlrpc.Proxy.callRemote error handling behavior

Author: exarkun
Reviewer: itamarst
Fixes [#10568](https://github.com/twisted/twisted/issues/10568)

This changes Proxy to propagate all errors from xmlrpclib.loads to the errback of
the Deferred which callRemote returns, rather than only xmlrpclib.Faults.

It also adds a unit test for this behavior.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
